### PR TITLE
Allow debug printing of the submission queue

### PR DESCRIPTION
**Pull request by zethra**
When debugging programs using io_uring it can be useful to check the
state of the submission queue. To make that easier I've added an
implementation of Debug on Entry and SubmissionQueue.

I'm not 100% sure I implemented this correctly so I'd appreciate a code review. I also added a test that just prints the queue full and then empty. I'm not sure if a different or more comprehensive testing method is desired.

### CODE REVIEW

**Comment by zethra** [comment]
It may also be desirable to implement Debug on other types if this is accepted.

File: src/squeue.rs [hunk 1]
```diff
@@ -297,3 +297,27 @@ impl Display for PushError {
...
+    }
+}
+
+impl Debug for SubmissionQueue<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let mut d = f.debug_list();
+        let mut pos = self.head;
+        while pos != self.tail {
+            let entry =
+                Entry(unsafe { *self.queue.sqes.add((pos & self.queue.ring_mask) as usize) });
```
Comment (quininer) [inline src/squeue.rs:317]: we can get `&Entry` directly, thus avoiding a copy?

**Review by quininer — CHANGES_REQUESTED** [review]
I don’t like adding too much `impl Debug`, but if it helps, I’m not against it.

File: src/squeue.rs (hunk 1 above)
Comment (zethra) [reply src/squeue.rs]: Yes, done.
